### PR TITLE
DQM: Added new package MuonMonitor for cosmic muons

### PR DESCRIPTION
**Pull request by calderona**
#### PR description:

Added new package on DQM online for cosmic muon monitoring, under DQM/MuonMonitor and also a new client under DQM/Integration/clients/

#### PR validation:

tested locally. All the code compile, runs and histograms appears on DQM GUI.

### CODE REVIEW

**Comment by calderona** [comment]
Already requested a PR for https://github.com/cms-sw/cms-bot/blob/master/categories_map.py to assign category

https://github.com/cms-sw/cms-bot/pull/1214

**Comment by jfernan2** [comment]
please test

**Comment by smuzaffar** [comment]
@calabria , the new package DQM/MuonMonitor just has one header file in interface and src dirs and it generates a plugin. Why not add these files in any existing package's plugins directory?

**Comment by fabiocos** [comment]
@calderona @smuzaffar I was looking at this today, I agree with @smuzaffar that this looks slim for a package: is an extension of it is foreseen in next future? We have other DQM packages specialized by subdetector, not obvious where else this could be merged.

**Comment by jfernan2** [comment]
I support @fioriNTU arguments. Thanks

File: DQM/MuonMonitor/BuildFile.xml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+<use   name="Geometry/CSCGeometry"/>
```
Comment (fabiocos) [inline DQM/MuonMonitor/BuildFile.xml]: @calderona are all these dependencies really needed? Where are digis rechits and geometry used for instance?

**Comment by fabiocos** [comment]
@jfernan2 @fioriNTU although not ideal, a package with a single module could still be acceptable (we are pushing for a better structure). But it would be good to structure it with a plugin directory, move the BuildFile there and possibly merge header and source (no reason to have them separated I would say). I know that many other DQM packages share this kind of structure, but in principle this is something that would be better to modify, @smuzaffar is also suggesting to go in this direction

**Comment by boudoul** [comment]
Hello @calderona , can we make the title more explicit please ?  I don't think when this title will be made public in the release notes it will be easy to understand the purpose ... thanks ( my comment is also valid for the backport  #28251  )

**Comment by jfernan2** [comment]
please test

**Comment by jfernan2** [comment]
+1

File: DQM/MuonMonitor/BuildFile.xml (hunk 1 above)
Comment (calderona) [reply DQM/MuonMonitor/BuildFile.xml]: Hi @fabiocos 
sorry, I thought I have answered this question already. I modified the BuildFile: https://github.com/cms-sw/cmssw/blob/3c1ab03ea63d4662d5488e7ac2d9d8573931c2e3/DQM/MuonMonitor/plugins/BuildFile.xml
It is true still there are some files for digi, so I will remove them.

**Comment by fabiocos** [comment]
please test

File: DQM/MuonMonitor/plugins/BuildFile.xml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+<use   name="RecoMuon/TrackingTools"/>
+<use   name="DataFormats/FEDRawData"/>
+<use   name="DataFormats/CSCRecHit"/>
+<use   name="DataFormats/DTRecHit"/>
+<use   name="CondFormats/DTObjects"/>
+<use   name="CondFormats/CSCObjects"/>
+<use   name="CondFormats/DataRecord"/>
+<use   name="CommonTools/TriggerUtils"/>
+<use name="DQM/MuonMonitor"/>
+</library>
```
Comment (fabiocos) [inline DQM/MuonMonitor/plugins/BuildFile.xml]: @calderona scram fires a warning here

```
****WARNING: Invalid tool DQM/MuonMonitor. Please fix src/DQM/MuonMonitor/plugins/BuildFile.xml file.
```

Comment (calderona) [reply DQM/MuonMonitor/plugins/BuildFile.xml]: Fixed.

**Comment by fabiocos** [comment]
@calderona to try to converge without too many trials: I have run

```
createBuildFile.pl --dir DQM/MuonMonitor/plugins
```

and got as an output

```
<library name="DQMMuonMonitorPlugins" file="*.cc">
 <use name="DQMServices/Core"/>
 <use name="DataFormats/Common"/>
 <use name="DataFormats/MuonReco"/>
 <use name="DataFormats/TrackReco"/>
 <use name="FWCore/Framework"/>
 <use name="FWCore/MessageLogger"/>
 <use name="FWCore/ParameterSet"/>
 <use name="FWCore/PluginManager"/>
 <use name="FWCore/ServiceRegistry"/>
 <use name="FWCore/Utilities"/>
 <use name="RecoMuon/TrackingTools"/>
 <use name="roofitcore"/>
 <flags EDM_PLUGIN="1"/>
</library>
```

The presence of ```roofitcore``` makes me suspicious that some of the included headers are not really needed (i.e. ```TMath``` and ```TList```, where are they used?)

@smuzaffar FYI as we discussed earlier today

**Comment by calderona** [comment]
> @calderona to try to converge without too many trials: I have run
> 
> ```
> createBuildFile.pl --dir DQM/MuonMonitor/plugins
> ```
> 
> and got as an output
> 
> ```
> <library name="DQMMuonMonitorPlugins" file="*.cc">
>  <use name="DQMServices/Core"/>
>  <use name="DataFormats/Common"/>
>  <use name="DataFormats/MuonReco"/>
>  <use name="DataFormats/TrackReco"/>
>  <use name="FWCore/Framework"/>
>  <use name="FWCore/MessageLogger"/>
>  <use name="FWCore/ParameterSet"/>
>  <use name="FWCore/PluginManager"/>
>  <use name="FWCore/ServiceRegistry"/>
>  <use name="FWCore/Utilities"/>
>  <use name="RecoMuon/TrackingTools"/>
>  <use name="roofitcore"/>
>  <flags EDM_PLUGIN="1"/>
> </library>
> ```
> 
> The presence of `roofitcore` makes me suspicious that some of the included headers are not really needed (i.e. `TMath` and `TList`, where are they used?)
> 
> @smuzaffar FYI as we discussed earlier today

Hi, 

TMath is used in the Chi2 probability: https://github.com/cms-sw/cmssw/pull/28244/files#diff-b6324f9e5797d9b81c67855527a5238bR253

TList, has been removed. In any case, I continue getting 'roofitcore' in the BuildFile when doing createBuildFile.pl --dir DQM/MuonMonitor/plugins .

**Comment by fabiocos** [comment]
@calderona ok, because of TMath at this point I would say. Not sure why ```roofitcore``` is picked, as ```rootmathcore``` is enough to get a compiling version. Please use the BuildFile provided by the script and let's integrate this PR

**Comment by calderona** [comment]
> @calderona ok, because of TMath at this point I would say. Not sure why `roofitcore` is picked, as `rootmathcore` is enough to get a compiling version. Please use the BuildFile provided by the script and let's integrate this PR

Ok, I have fixed the BuildFile.

**Comment by jfernan2** [comment]
please test

**Comment by fabiocos** [comment]
@calderona @jfernan2 I am ok with this PR in its present form. Is it needed for the last MWGR?

**Comment by fioriNTU** [comment]
@fabiocos yes, we would like to test this PR in Online for the next MWGR. We can also deply this single PR on the Online system later, but if it is integrated in some CMSSW release would be of course better.

**Comment by fabiocos** [comment]
@fioriNTU very well, if someone from DQM signs it it would be even better :-)

**Comment by jfernan2** [comment]
+1

**Comment by fioriNTU** [comment]
+1

**Comment by fabiocos** [comment]
+1

**Comment by boudoul** [comment]
Hello @calderona  , @kmaeshima,  @jfernan2, @fioriNTU, shouldn't we get a backport in 10_6 ? Sorry If i miss something  . Thanks

**Comment by fioriNTU** [comment]
There is already a backport https://github.com/cms-sw/cmssw/pull/28251 , we need to ask for a merge at the Orp today

**Comment by boudoul** [comment]
ok thanks

**Pull request merged**